include/kodlab_mjbots_sdk/pid.h

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file pid.h
+ * @author Chandravaran Kunjeti (kunjeti@seas.upenn.edu)
+ * @brief Class implementation of PID
+ * @date 9/30/2022
+ *
+ * @copyright 2022 The Trustees of the University of Pennsylvania.
+ * All rights reserved.
+ *
+ */
+
+#pragma once
+
+#include <math.h>
+
+#include <vector>
+
+#define HIGH_SATURATION_INDEX_ 0
+#define LOW_SATURATION_INDEX_ 0
```

Comment:
What is the point of these saturation indices?  The current method seems convoluted, why not simply store the limits in appropriately labelled members?

Reply:
So I am planning on adding checks on the limits. So that when the user uses the code properly. I notice I made mistake here and didn't change the index value for the low saturation value. Also I defined these constant variables so the code could be more understandable. If the code just contained 0,1 a question may arise in the future as to what is in that index.

Reply:
As long as the `saturation_` member documentation appropriately explains the indexing then this isn't totally necessary, but still good to have if you really want to be explicit.  

If you do want to add these constants, they should not be macros polluting the global namespace.  Instead, add them as namespaced constants within the class itself.  Since these values are fixed at compile time, they can be declared `constexpr`.  Further, they should be declared `inline` as described [here](https://stackoverflow.com/a/56950545).  

The benefit of doing it this way is that the constant is only accessible in the relevant namespace, and not everywhere that the header file is included.